PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx

```diff
@@ -0,0 +1,1111 @@
+
+      // computing cosPA
+      double cpaV0 = casc.v0cosPA(collision.posX(), collision.posY(), collision.posZ());
+      double cpaCharmBaryon = RecoDecay::cpa(pvCoord, coordVtxCharmBaryon, pVecCharmBaryon);
+      double cpaCasc = casc.casccosPA(collision.posX(), collision.posY(), collision.posZ());
+      double cpaxyV0 = RecoDecay::cpaXY(pvCoord, vertexV0, pVecV0);
+      double cpaxyCharmBaryon = RecoDecay::cpaXY(pvCoord, coordVtxCharmBaryon, pVecCharmBaryon);
+      double cpaxyCasc = RecoDecay::cpaXY(pvCoord, vertexCasc, pVecCasc);
+
+      // computing decay length and ctau
+      double decLenCharmBaryon = RecoDecay::distance(pvCoord, coordVtxCharmBaryon);
+      double decLenCascade = RecoDecay::distance(coordVtxCharmBaryon, vertexCasc);
+      double decLenV0 = RecoDecay::distance(vertexCasc, vertexV0);
+
+      double phiCharmBaryon, thetaCharmBaryon;
+      getPointDirection(std::array{primaryVertex.getX(), primaryVertex.getY(), primaryVertex.getZ()}, coordVtxCharmBaryon, phiCharmBaryon, thetaCharmBaryon);
+      auto errorDecayLengthCharmBaryon = std::sqrt(getRotatedCovMatrixXX(primaryVertex.getCov(), phiCharmBaryon, thetaCharmBaryon) + getRotatedCovMatrixXX(covVtxCharmBaryon, phiCharmBaryon, thetaCharmBaryon));
+      auto errorDecayLengthXYCharmBaryon = std::sqrt(getRotatedCovMatrixXX(primaryVertex.getCov(), phiCharmBaryon, 0.) + getRotatedCovMatrixXX(covVtxCharmBaryon, phiCharmBaryon, 0.));
+
+      double ctOmegac = RecoDecay::ct(pVecCharmBaryon, decLenCharmBaryon, massOmegacFromPDG);
+      double ctXic = RecoDecay::ct(pVecCharmBaryon, decLenCharmBaryon, massXicFromPDG);
+      double ctCascade = 0.;
+      if constexpr (decayChannel == hf_cand_casc_lf::DecayType2Prong::XiczeroOmegaczeroToXiPi) {
+        ctCascade = RecoDecay::ct(pVecCasc, decLenCascade, massXiFromPDG);
+      } else {
+        ctCascade = RecoDecay::ct(pVecCasc, decLenCascade, massOmegaFromPDG);
+      }
+      double ctV0 = RecoDecay::ct(pVecV0, decLenV0, massLambdaFromPDG);
+
+      // computing eta
+      double pseudorapCharmBaryon = RecoDecay::eta(pVecCharmBaryon);
+      double pseudorapCascade = RecoDecay::eta(pVecCasc);
+      double pseudorapV0 = RecoDecay::eta(pVecV0);
+
+      // DCA between daughters
+      float dcaCascDau = casc.dcacascdaughters();
+      float dcaV0Dau = casc.dcaV0daughters();
+      float dcaCharmBaryonDau = std::sqrt(df.getChi2AtPCACandidate());
+
+      // set hfFlag
+      int hfFlag = 0;
+      if constexpr (decayChannel == hf_cand_casc_lf::DecayType2Prong::XiczeroOmegaczeroToXiPi) {
+        hfFlag = 1 << aod::hf_cand_xic0omegac0::DecayType::DecayToXiPi;
+      }
+      if constexpr (decayChannel == hf_cand_casc_lf::DecayType2Prong::OmegaczeroToOmegaPi) {
+        hfFlag = 1 << aod::hf_cand_xic0omegac0::DecayType::DecayToOmegaPi;
+      } else {
+        hfFlag = 1 << aod::hf_cand_xic0omegac0::DecayType::DecayToOmegaK;
+      }
```

Comment:
This sets the wrong flag if it is `XiczeroOmegaczeroToXiPi` since it enters both in the first `if` and in the second `else`
```suggestion
      int hfFlag = 1 << decayChannel;
```
Moreover, since you anyway only store one channel per table, is it really necessary to have this flag stored in the output table?